crates/task-impls/src/quorum_proposal.rs

```diff
@@ -0,0 +1,541 @@
+                    event_receiver,
+                    event_sender,
+                    event.clone(),
+                );
+            }
+            HotShotEvent::ViewSyncFinalizeCertificate2Recv(view_sync_finalize_cert) => {
+                let view = view_sync_finalize_cert.view_number;
+                if view < self.latest_proposed_view {
+                    debug!(
+                        "View sync certificate is from an old view {:?}",
+                        view.clone()
+                    );
+                    return;
+                }
+
+                if !validate_view_sync_finalize_certificate(
+                    view_sync_finalize_cert.clone(),
+                    event_sender.clone(),
+                ) {
+                    return;
+                }
+
+                self.create_dependency_task_if_new(view, event_receiver, event_sender, event);
+            }
+            HotShotEvent::QuorumProposalRecv(proposal, _sender) => {
+                let view = proposal.data.get_view_number();
+                if view < self.latest_proposed_view {
+                    debug!("Proposal is from an older view {:?}", proposal.data.clone());
+                    return;
+                }
+
+                debug!(
+                    "Received Quorum Proposal for view {}",
+                    *proposal.data.view_number
+                );
+
+                // stop polling for the received proposal
+                self.quorum_network
+                    .inject_consensus_info(ConsensusIntentEvent::CancelPollForProposal(
+                        *proposal.data.view_number,
+                    ))
+                    .await;
+
+                if !validate_quorum_proposal(proposal.clone(), event_sender.clone()) {
+                    return;
+                }
+                broadcast_event(
+                    Arc::new(HotShotEvent::QuorumProposalValidated(proposal.data.clone())),
+                    &event_sender.clone(),
+                )
+                .await;
+
+                self.create_dependency_task_if_new(
+                    view,
+                    event_receiver,
+                    event_sender,
+                    event.clone(),
+                );
+            }
+            HotShotEvent::QuorumProposalDependenciesValidated(view) => {
+                debug!("All proposal dependencies verified for view {:?}", view);
+                if !self.update_latest_proposed_view(*view).await {
+                    debug!("proposal not updated");
+                    return;
+                }
+            }
+            _ => {}
+        }
+    }
+}
+
+impl<TYPES: NodeType, I: NodeImplementation<TYPES>> TaskState
+    for QuorumProposalTaskState<TYPES, I>
+{
+    type Event = Arc<HotShotEvent<TYPES>>;
+    type Output = ();
+    fn filter(&self, event: &Arc<HotShotEvent<TYPES>>) -> bool {
+        !matches!(
+            event.as_ref(),
+            HotShotEvent::QuorumProposalRecv(_, _)
```

Comment:
@bfish713 @elliedavidson @shenkeyao am I missing any events? This appears to be all of them, and I don't see any others in the consensus task that are being used to initiate a proposal.

Reply:
Yeah, I think this is all.